src/Tribe/Admin/Activation_Page.php

```diff
@@ -0,0 +1,283 @@
+	 * a higher version).
+	 */
+	protected function log_display_of_message_page() {
+		Tribe__Settings_Manager::set_option( 'last-update-message-' . $this->args['slug'], $this->args['version'] );
+	}
+
+	/**
+	 * The previous_ecp_versions option will be empty or set to 0
+	 * if the current version is the first version to be installed.
+	 *
+	 * @return bool
+	 * @see Tribe__Events__Main::maybeSetTECVersion()
+	 */
+	protected function is_new_install() {
+		$previous_versions = Tribe__Settings_Manager::get_option( $this->args['version_history_slug'] );
+		return empty( $previous_versions ) || ( end( $previous_versions ) == '0' );
+	}
+
+	/**
+	 * Handles taking a user to a post-installation welcome page.
+	 */
+	protected function redirect_to_welcome_page() {
+		$url = $this->get_message_page_url( $this->welcome_slug );
+		wp_safe_redirect( $url );
+		exit();
+	}
+
+	/**
+	 * Handles taking the user to a post-update splash screen.
+	 *
+	 * Disused since TEC PR 88 (targeting Tribe__Events__Activation_Page,
+	 * which this class was derived from).
+	 *
+	 * @see https://github.com/moderntribe/the-events-calendar/pull/88
+	 *
+	 * @todo decide whether to reinstate or remove
+	 */
+	protected function redirect_to_update_page() {
+		$url = $this->get_message_page_url( $this->update_slug );
+		wp_safe_redirect( $url );
+		exit();
+	}
+
+	/**
+	 * Return the URL of the splash page.
+	 *
+	 * @param string $slug
+	 *
+	 * @return string
+	 */
+	protected function get_message_page_url( $slug ) {
+		$settings = Tribe__Settings::instance();
+		// get the base settings page url
+		$url  = apply_filters(
+			'tribe_settings_url',
+			add_query_arg( 'page', $settings->adminSlug, admin_url( 'edit.php' ) )
+		);
+		$url = esc_url_raw( add_query_arg( $slug, 1, $url ) );
+		return $url;
+	}
+
+	/**
+	 * Dynamically registers the splash page when required.
+	 */
+	public function register_page() {
+		if ( isset( $_GET[ $this->welcome_slug ] ) ) {
+			$this->disable_default_settings_page();
+			add_action( Tribe__Settings::instance()->admin_page, array( $this, 'display_page' ) );
+		} elseif ( isset( $_GET[ $this->update_slug ] ) ) {
+			$this->disable_default_settings_page();
+			add_action( Tribe__Settings::instance()->admin_page, array( $this, 'display_page' ) );
+		}
+	}
+
+	/**
+	 * Deactivates the regular settings screen (the splash screen will display
+	 * in the Events > Settings slot instead, for this request only).
+	 */
+	protected function disable_default_settings_page() {
+		remove_action( Tribe__Settings::instance()->admin_page, array( Tribe__Settings::instance(), 'generatePage' ) );
+	}
+
+	/**
+	 * Prints the splash screen.
+	 *
+	 * @param string $context
+	 */
+	public function display_page( $context = 'welcome' ) {
+		do_action( 'tribe_settings_top' );
+
+		$context = isset( $_GET[ $this->welcome_slug ] ) ? 'welcome': 'update';
+		$title   = esc_html( @$this->args[ $context . '_page_title'] );
```

Comment:
It may not be the very best option :wink:

It's a precaution; in general and in our own internal usage it shouldn't be needed. As the method is public, though (has to be, it's a callback), I added it as a precaution.

We can add extra checks though, what do you think?